js/bars.js

```diff
@@ -1027,9 +1042,28 @@ historian.Bars.prototype.filter_ = function(data, callback) {
  */
 historian.Bars.prototype.filterByAppUid_ = function(data, uid) {
   return this.filter_(data, function(entry) {
-    var value = entry.value;
-    var valueMap = value.split(",");
-    var eventUID = valueMap[6];
+    // Only keep the entries that corresponds to the given uid.
+    var valueMap = entry.value.split(",");
+    var eventUID;
+    switch(valueMap[0]) {
+      // For error entry, the value field starts with an error tag, so we
+      // case on the error tag to get the uid.
```

Comment:
switch on the error tag instead of case

Reply:
I am not sure what you mean here. Should I just use if conditions instead of switch cases?

Reply:
In the comment, replace case with switch.